csrc/custom_all_reduce.cuh

```diff
@@ -220,13 +273,13 @@ __global__ void __launch_bounds__(512, 1)
   // note: we don't reorder the address so the accumulation order is the same
   // for all ranks, ensuring bitwise identical results
   auto dp = *_dp;
-  multi_gpu_barrier<ngpus, true>(sg, self_sg, rank);
+  start_sync<ngpus>(sg, self_sg, rank);
   // do the actual reduction
   for (int idx = blockIdx.x * blockDim.x + threadIdx.x; idx < size;
        idx += gridDim.x * blockDim.x) {
     ((P*)result)[idx] = packed_reduce<P, ngpus, A>((const P**)&dp.ptrs[0], idx);
   }
-  multi_gpu_barrier<ngpus, false>(sg, self_sg, rank);
+  end_sync<ngpus, true>(sg, self_sg, rank);
```

Comment:
Could you explain why you changed the boolean value here?

Reply:
The template parameter meaning has changed. It is opposite now. "True" used to mean here that we want to do a memory fence, now it means that we don't have to do any additional thread sync as this is the end of the kernel.

Reply:
Small nit on naming: The names `start_sync` and `end_sync` makes me think that we're starting and ending the synchronization, and that both are needed to make a barrier -- but IIUC, 
it should be read as `sync_at_start` and `sync_at_end`, so consider renaming the functions for clarity